### PR TITLE
Examples: Add support for overriding lorawan keys during build via LORA_APPKEY, LORA_DEVEUI.. values

### PR DESCRIPTION
**Pull request by plaes**
Implement lorawan key customization via environment variables (for nrf52840):
`LORA_APPEUI= LORA_APPKEY= LORA_DEVEUI=1122334455667788 cargo build --release`

### CODE REVIEW

**Review by lucasgranberg — DISMISSED** [review]

**Comment by lthiery** [comment]
What about providing this as a util in the lorawan module? You could call the util from build.rs.

**Comment by lthiery** [comment]
Come to think of it, providing this as a util in the lorawan module will be difficult because this (and build.rs) can depend on std but lorawan can't. 

We could create lorawan-util crates that has std... but I think this is also fine the way it is.

File: examples/nrf52840/build.rs [hunk 1]
```diff
@@ -10,14 +10,77 @@
 
 use std::env;
 use std::fs::File;
-use std::io::Write;
+use std::io::{BufWriter, Write};
 use std::path::PathBuf;
 
+fn hex_to_bytes(s: &str) -> Option<Vec<u8>> {
```
Comment (lthiery) [inline examples/nrf52840/build.rs:16]: hex::encode?

File: examples/nrf52840/build.rs [hunk 2]
```diff
@@ -10,14 +10,77 @@
...
+        write!(
+            &mut file,
+            "{}",
+            format_args!(
+                "\
+            // Generated by build.rs\n\
+            const DEVEUI: Option<[u8; 8]> = {};\n\
+            const APPEUI: Option<[u8; 8]> = {};\n\
+            const APPKEY: Option<[u8; 16]> = {};\n",
+                parse_lorawan_id(option_env!("LORA_DEVEUI"), "LORA_DEVEUI", 8).unwrap_or("None".to_string()),
```
Comment (lthiery) [inline examples/nrf52840/build.rs:74]: perhaps implement `DevEui/AppEui/AppKey::from_str`? Feature-gated behind std being enabled in lorawan-encoding.

**Comment by lthiery** [comment]
Actually, I was wrong above. `build-dependencies` could have std enabled for `lorawan-device` which could provide this utility without having std enabled in the project itself. Then that could enable the `from_str` from `lorawan-encoding` (see review)?

**Review by lthiery — APPROVED** [review]
Some non-blocking suggestions.

**Comment by lthiery** [comment]
I also see that we hand-rolled the [hex encoding here](https://github.com/lora-rs/lora-rs/blob/main/lorawan-encoding/src/extra/to_string.rs) in lorawan-encoding. I've corrected there [here](https://github.com/lora-rs/lora-rs/commit/67bf7eae2b4322ef59e6b136806fa7fb97176dec).

**Comment by lthiery** [comment]
I went ahead and just made the PR for lorawan-encoding: https://github.com/lora-rs/lora-rs/pull/234

File: examples/nrf52840/build.rs (hunk 1 above)
Comment (plaes) [reply examples/nrf52840/build.rs:16]: Yeah, I initially didn't want to include extra dependency for just a single semi-trivial function (`left-pad` :dagger: )..

**Comment by lucasgranberg** [comment]
just for consideration the hex decoding can be done at compile time without custom build scripts by using the hex-literal crate. https://crates.io/crates/hex-literal

**Comment by plaes** [comment]
So, I basically ran into a headache and current version is good enough.
I can easily use the `build.rs` to parse the DevEui and friends, but in the end I would need to somehow "tell" rust to use this value in some way, which means I would have to either convert it back to array syntax or find something better.
This is how it works for now:
1. In generated file I have: `const DEVEUI: Option<[u8; 8]> = None;`
2. In lorawan example code  I need to match the types so I can override: `deveui: DevEui::from(DEVEUI.unwrap_or(DEFAULT_DEVEUI)),`

(Also, `from_str()` requires pulling in `use core::str::FromStr;` and its a lot easier to type out zeroes as array than into hex string..)

**Pull request merged**